ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java

```diff
@@ -97,18 +120,55 @@ private Transactional getTransactional(InvocationContext ic) {
             return transactional;
         }
 
-        // see if the target is a stereotype
-        for (Annotation annotation : interceptedBean.getBeanClass().getAnnotations()) {
+        // check existence of the stereotype on method
+        Transactional transactionalMethod = getTransactionalAnnotationRecursive(ic.getMethod().getAnnotations());
```

Comment:
You can make use of `AnnotatedMethod` here -> `AnnotatedType<T>#getMethods()` and filter it. This will make sure you have the actual annotations as people might have changed this via extensions (as they can with whole AT).